aws_sagemaker_studio/frameworks/pytorch_horovod_mpi/pytorch_horovod_mpi.ipynb

```diff
@@ -0,0 +1,407 @@
+    "\n",
+    "In this notebook, we:\n",
+    "\n",
+    "* Upload a data set to S3\n",
+    "* Train a simple neural network with Horovod\n",
+    "* Deploy a model to a SageMaker endpoint"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "For this example, we use the MNIST data set. MNIST is a widely used dataset for handwritten digit classification. It consists of 70,000 labeled 28x28 pixel grayscale images of hand-written digits. The dataset is split into 60,000 training images and 10,000 test images. There are 10 classes (one for each of the 10 digits)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Downloading the Data\n",
+    "\n",
+    "The MNIST data set can be downloaded with the `torchvision.datasets` package."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set the path we can find the data files that go with this notebook\n",
+    "%cd /root/amazon-sagemaker-examples/aws_sagemaker_studio/frameworks/pytorch_horovod_mpi"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "import torchvision\n",
+    "\n",
+    "# ipywidgets is required to display loading bars in Jupyter. Because ipywidgets is not installed\n",
+    "# in the container by default, we're temporarily disabling the loading bar.\n",
+    "def gen_bar_updater():\n",
+    "    return lambda count, block_size, total_size: None\n",
+    "torchvision.datasets.utils.gen_bar_updater = gen_bar_updater\n",
+    "\n",
+    "dataset = torchvision.datasets.MNIST(\n",
+    "    root='./data',\n",
+    "    transform=torchvision.transforms.ToTensor(),\n",
+    "    download=True,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Previewing the Data\n",
+    "\n",
+    "Let's preview what the input images look like. We can do this using `matplotlib`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "dataloader = torch.utils.data.DataLoader(dataset, batch_size=4)\n",
+    "images, labels = next(iter(dataloader))\n",
+    "grid = torchvision.utils.make_grid(images)\n",
+    "plt.imshow(grid.permute(1, 2, 0))  \n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Uploading the Data\n",
+    "\n",
+    "We use the `sagemaker.Session.upload` function to upload our datasets to an S3 location. The return value, `inputs`, identifies the location of our data.\n",
+    "\n",
+    "To upload our data to S3, we need to specify an S3 bucket and prefix. These should be within the same region as the Notebook Instance, training, and hosting."
```

Comment:
s/Notebook Instance/Notebook

(since it's Studio, it's a different entity from the legacy "notebook instances")